sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/JoinEstimation.scala

```diff
@@ -231,12 +231,11 @@ case class JoinEstimation(join: Join) extends Logging {
       max: Option[Any]): (BigInt, ColumnStat) = {
     val leftKeyStat = leftStats.attributeStats(leftKey)
     val rightKeyStat = rightStats.attributeStats(rightKey)
-    val maxNdv = leftKeyStat.distinctCount.get.max(rightKeyStat.distinctCount.get)
+    val newNdv = leftKeyStat.distinctCount.get.min(rightKeyStat.distinctCount.get)
     // Compute cardinality by the basic formula.
-    val card = BigDecimal(leftStats.rowCount.get * rightStats.rowCount.get) / BigDecimal(maxNdv)
+    val card = BigDecimal(leftStats.rowCount.get * rightStats.rowCount.get) / BigDecimal(newNdv)
```

Comment:
I don't see too much explanation why this is a better estimation in the PR description or in code comment. Could you add some?

Roughly looks like this change tries to overestimate cardinality instead underestimating previously, so it lowers the chance to broadcast join table.

Before this, if the join table fails to broadcast, user can disable broadcast to run the query. If after this change Spark doesn't broadcast the join table (as estimated states are higher than threshold), user may not be able to broadcast it (as the threshold is a hard limit).